File: test_loop.c
// Calculate the sum of first n natural numbers
int sum_iterative(int n) {
    int result = 0;
    for (int i = 1; i <= n; i++) {
        result += i;
    }
    return result;
}

// Calculate the sum of first n natural numbers using while loop
int sum_while(int n) {
    int result = 0;
    int i = 1;
    while (i <= n) {
        result += i;
        i++;
    }
    return result;
}

int main() {
    int n = 10;
    
    // Test for loop
    int for_result = sum_iterative(n);
    printf("Sum of first %d numbers using for loop: %d\n", n, for_result);
    
    // Test while loop
    int while_result = sum_while(n);
    printf("Sum of first %d numbers using while loop: %d\n", n, while_result);
    
    // Test if-else
    if (for_result == while_result) {
        printf("Both methods give the same result!\n");
    } else {
        printf("There's a bug in one of the implementations.\n");
    }
    
    return 0;
}

File: test_result/test_loop.py
from dataclasses import dataclass, field
from typing import List

# Global variables

# Functions
def sum_iterative(n: int) -> int:
    result: int = 0
    i: int = 1
    while (i <= n):
        result = (result + i)
        i = (i + 1)
    return result

def sum_while(n: int) -> int:
    result: int = 0
    i: int = 1
    while (i <= n):
        result = (result + i)
        i = (i + 1)
    return result

def main() -> int:
    n: int = 10
    for_result: int = sum_iterative(n)
    print(f"Sum of first {n} numbers using iterative method: {for_result}\n")
    while_result: int = sum_while(n)
    print(f"Sum of first {n} numbers using while loop: {while_result}\n")
    if (for_result == while_result):
        print(f"Both methods give the same result!\n")
    else:
        print(f"There's a bug in one of the implementations.\n")
    return 0

if __name__ == "__main__":
    main()
